src/main/java/uk/gov/hmcts/reform/hmc/interceptors/HeaderProcessor.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.hmc.interceptors;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.hmcts.reform.hmc.ApplicationParams;
+import uk.gov.hmcts.reform.hmc.config.UrlManager;
+import uk.gov.hmcts.reform.hmc.service.common.OverrideAuditService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class HeaderProcessor implements HandlerInterceptor {
+
+    private final ApplicationParams params;
+    private final UrlManager roleAssignmentUrlManager;
+    private final UrlManager dataStoreUrlManager;
+
+    private final OverrideAuditService overrideAuditService;
+
+    /**
+     * Check if role assignment and/or ccd data store url headers are present in the request.
+     * If any of the headers are present, set the actual host to the value of the header.
+     * @param request current HTTP request
+     * @param response current HTTP response
+     * @param handler chosen handler to execute, for type and/or instance evaluation
+     * @return boolean
+     * @throws Exception – in case of errors
+     */
+    @Override
+    public boolean preHandle(
+        HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        if (params.isHmctsDeploymentIdEnabled()) {
```

Comment:
maybe check it is not prod too? Deployment id should never be in prod so it's fine but. if it ever is, it'll be a security flaw.